src/php/AbstractPhelConfig.php

```diff
@@ -4,36 +4,29 @@
 
 namespace Phel;
 
-use RuntimeException;
-
-abstract class AbstractPhelConfig
+use Gacela\Framework\AbstractConfig;
+use Gacela\Framework\Config;
+use function json_decode;
+
+/**
+ * This is a layer on top of Gacela config, to get the values from the composer.json.
+ */
+abstract class AbstractPhelConfig extends AbstractConfig
 {
-    private array $config;
-
-    public function __construct(string $rootDir)
-    {
-        $composerContent = file_get_contents($rootDir . 'composer.json');
-        if (!$composerContent) {
-            throw new RuntimeException('Cannot read composer.json in: ' . $rootDir);
-        }
-
-        $composerData = \json_decode($composerContent, true);
-        if (!$composerData) {
-            throw new RuntimeException('Cannot read composer.json in: ' . $rootDir);
-        }
-
-        if (!isset($composerData['extra']['phel'])) {
-            throw new RuntimeException('No Phel configuration found in composer.json');
-        }
-
-        $this->config = $composerData['extra']['phel'];
-    }
+    private array $composerJson = [];
 
     /**
-     * @return mixed|null
+     * @override to load the config from the composer.json
+     *
+     * @param null|mixed $default
      */
-    public function get(string $key)
+    protected function get(string $key, $default = null)
     {
-        return $this->config[$key] ?? null;
+        if (empty($this->composerJson)) {
+            $composerJsonPath = Config::getApplicationRootDir() . DIRECTORY_SEPARATOR . 'composer.json';
+            $this->composerJson = json_decode(file_get_contents($composerJsonPath), true);
```

Comment:
Should this handle error like,
* composer.json is not readable
* composer.json is not parsable
?

Reply:
Good point. Yes, I will verify this and throw a RuntimeException giving some hints in case of error.

Reply:
Done: https://github.com/phel-lang/phel-lang/pull/253/commits/54a560bb95d0ca1cc7dbf500beb8fe2fea97df3a